api/api/utils/moderation_lock.py

```diff
@@ -1,34 +1,41 @@
 import time
 
+import django_redis
 import structlog
-from django_redis import get_redis_connection
 from redis import Redis
+from redis.exceptions import ConnectionError
 
 
 REPORT_LOCK_PREFIX = "soft_lock_report"
 MODERATOR_LOCK_PREFIX = "soft_lock_moderator"
 PREFIXES = [REPORT_LOCK_PREFIX, MODERATOR_LOCK_PREFIX]
 
 logger = structlog.get_logger(__name__)
-redis: Redis = get_redis_connection("default")
 
 
 class LockManager:
     def __init__(self, media_type):
         self.media_type = media_type
+        redis: Redis = django_redis.get_redis_connection("default")
+        try:
+            redis.ping()
+            self.redis = redis
+        except ConnectionError:
+            logger.error("Redis connection failed")
+            self.redis = None
```

Comment:
Just a nit, but can you move this into a method `get_redis` that retrieves and tests it when used, rather than in the constructor? Otherwise, this constructor makes a blocking network call, which is rather unexpected.

This is more important once we move to use an async Redis client, because constructors cannot be asynchronous. In that case, each of the methods that use redis would already be asynchronous anyway.

Just a nit though, not strictly necessary for this PR, but something we know we'll need to refactor one day.

Reply:
I've refactored this in 2e7c59e so that `self.redis` is no longer a thing in `LockManager`.